csi/controller_server.go

```diff
@@ -632,6 +637,9 @@ func (cs *ControllerServer) waitForVolumeState(volumeID string, state types.Volu
 				return notFoundReturn
 			}
 			if existVol.State == string(state) {
+				if state == types.VolumeStateAttached && existVol.Controllers[0].Endpoint == "" {
```

Comment:
this is sort of a hack, but okay for now.
I have a refactor for this state checking function in my branch.